edgelet/support-bundle/src/shell_util.rs

```diff
@@ -0,0 +1,251 @@
+    let mut inspect = Command::new("docker");
+    inspect.args(&["network", "ls"]);
+    inspect.args(&["--format", "{{.Name}}"]);
+    let inspect = inspect.output().await;
+
+    let result = if let Ok(result) = inspect {
+        if result.status.success() {
+            String::from_utf8_lossy(&result.stdout).to_string()
+        } else {
+            println!(
+                "Could not find network names: {}",
+                String::from_utf8_lossy(&result.stderr)
+            );
+            "azure-iot-edge".to_owned()
+        }
+    } else {
+        println!("Could not find network names: {}", inspect.err().unwrap());
+        "azure-iot-edge".to_owned()
+    };
+
+    let result = result.lines().map(String::from).collect();
+    Ok(result)
+}
+
+pub async fn write_network_inspect<W>(
+    network_name: &str,
+    zip_writer: &mut ZipWriter<W>,
+    file_options: &FileOptions,
+    verbose: bool,
+) -> Result<(), Error>
+where
+    W: Write + Seek,
+{
+    print_verbose(
+        &format!("Running docker network inspect for {}", network_name),
+        verbose,
+    );
+    let mut inspect = Command::new("docker");
+
+    inspect.args(&["network", "inspect", &network_name, "-v"]);
+    let inspect = inspect.output().await;
+
+    let (file_name, output) = if let Ok(result) = inspect {
+        if result.status.success() {
+            (format!("network/{}.json", network_name), result.stdout)
+        } else {
+            (format!("network/{}_err.json", network_name), result.stderr)
+        }
+    } else {
+        let err_message = inspect.err().unwrap().to_string();
+        println!(
+            "Could not reach docker. Including error in bundle.\nError message: {}",
+            err_message
+        );
+        (
+            format!("network/{}_err_docker.txt", network_name),
+            err_message.as_bytes().to_vec(),
+        )
+    };
+
+    zip_writer
+        .start_file(file_name, *file_options)
+        .map_err(|err| Error::from(err.context(ErrorKind::SupportBundle)))?;
+
+    zip_writer
+        .write_all(&output)
+        .map_err(|err| Error::from(err.context(ErrorKind::SupportBundle)))?;
+
+    print_verbose(
+        &format!("Got docker network inspect for {}", network_name),
+        verbose,
+    );
+    Ok(())
+}
+
+pub async fn write_system_log<W>(
+    name: &str,
+    unit: &str,
+    log_options: &LogOptions,
+    zip_writer: &mut ZipWriter<W>,
+    file_options: &FileOptions,
+    verbose: bool,
+) -> Result<(), Error>
+where
+    W: Write + Seek,
+{
+    print_verbose(
+        format!("Getting system logs for {}", name).as_str(),
+        verbose,
+    );
+    let since_time: DateTime<Utc> = DateTime::from_utc(
+        NaiveDateTime::from_timestamp(log_options.since().into(), 0),
+        Utc,
+    );
+    let until_time: Option<DateTime<Utc>> = log_options
+        .until()
+        .map(|until| DateTime::from_utc(NaiveDateTime::from_timestamp(until.into(), 0), Utc));
+
+    #[cfg(unix)]
```

Comment:
Do we need this?

Reply:
yes, docker accepts local time

Reply:
But we're always on unix now, right? So do we need to specify cfg(unix)?